### PR TITLE
KOGITO-9695: SWF Editor - Refactor GWT components

### PR DESCRIPTION
**Pull request by treblereel**
jira: https://issues.redhat.com/browse/KOGITO-9695

### CODE REVIEW

**Comment by tiagobento** [comment]
@treblereel Please put the Jira in the PR title.

**Review by handreyrc — APPROVED** [review]
Hey @treblereel,

The code looks good to me. I stressed the swf editor and I couldn't find any issues.  

Thanks for your work here!

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramEditorTest.java [hunk 1]
```diff
@@ -359,16 +357,6 @@ public void testUpdateContent() {
         verify(diagramApi).setContentSuccess();
     }
 
-    @Test
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramEditorTest.java:362]: Why did you remove the test?

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/LienzoCanvasViewTest.java [hunk 2]
```diff
@@ -48,6 +49,8 @@
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+// TODO: J2CL migration - Drop the Ignore once 2 TODOs fixed
+@Ignore
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/LienzoCanvasViewTest.java:53]: Is it reported?

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/test/java/org/kie/workbench/common/stunner/client/widgets/canvas/DelegateLienzoPanelTest.java [hunk 3]
```diff
@@ -90,9 +92,10 @@ public void testDestroy() {
     }
 
     @Test
-    public void testAsWidget() {
-        Widget widget = mock(Widget.class);
-        when(delegate.asWidget()).thenReturn(widget);
-        assertEquals(widget, tested.asWidget());
+    @Ignore("throws CCE")
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/test/java/org/kie/workbench/common/stunner/client/widgets/canvas/DelegateLienzoPanelTest.java:95]: Is it reported?

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/proxies/ElementProxyTest.java [hunk 4]
```diff
@@ -120,6 +121,7 @@ public void testShapeBuilder() {
     }
 
     @Test
+    @Ignore("CCE")
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/proxies/ElementProxyTest.java:124]: Is it reported?

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/toolbox/actions/GeneralCreateNodeActionTest.java [hunk 5]
```diff
@@ -144,6 +145,7 @@ public void setUp() throws Exception {
     }
 
     @Test
+    @Ignore("CCE")
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/toolbox/actions/GeneralCreateNodeActionTest.java:148]: Is it reported?

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/toolbox/actions/MorphNodeToolboxActionTest.java [hunk 6]
```diff
@@ -156,6 +157,7 @@ public void testGlyph() {
     }
 
     @Test
+    @Ignore("CCE")
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/toolbox/actions/MorphNodeToolboxActionTest.java:160]: Is it reported?

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/test/java/org/kie/workbench/common/stunner/lienzo/primitive/AbstractDragProxyTest.java [hunk 7]
```diff
@@ -144,6 +145,7 @@ public void testOnMouseUpWhenProxyIsNotAttached() {
     }
 
     @Test
+    @Ignore("NoSuchMethod")
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/test/java/org/kie/workbench/common/stunner/lienzo/primitive/AbstractDragProxyTest.java:148]: Is it reported?

**Review by LuboTerifaj — COMMENTED** [review]
Hello @treblereel ,

I didn't find any issues.

Please have a look at my inline comments and let me know your thoughts.

I will approve the PR afterwards.

Thank you!

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramEditorTest.java (hunk 1 above)
Comment (treblereel) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramEditorTest.java:362]: I will restore this test once the ported regex is added to the swf. 
this once is gwt-specific and our goal to remove gwt deps at all
see:KOGITO-9756

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/LienzoCanvasViewTest.java (hunk 2 above)
Comment (treblereel) [reply packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/LienzoCanvasViewTest.java:53]: yeap, at KOGITO-9756

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/test/java/org/kie/workbench/common/stunner/client/widgets/canvas/DelegateLienzoPanelTest.java (hunk 3 above)
Comment (treblereel) [reply packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/test/java/org/kie/workbench/common/stunner/client/widgets/canvas/DelegateLienzoPanelTest.java:95]: at KOGITO-9756

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/toolbox/actions/GeneralCreateNodeActionTest.java (hunk 5 above)
Comment (treblereel) [reply packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/toolbox/actions/GeneralCreateNodeActionTest.java:148]: yes, at KOGITO-9756

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/proxies/ElementProxyTest.java (hunk 4 above)
Comment (treblereel) [reply packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/proxies/ElementProxyTest.java:124]: the same

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/test/java/org/kie/workbench/common/stunner/lienzo/primitive/AbstractDragProxyTest.java (hunk 7 above)
Comment (treblereel) [reply packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/test/java/org/kie/workbench/common/stunner/lienzo/primitive/AbstractDragProxyTest.java:148]: yeap, KOGITO-9756

**Review by LuboTerifaj — APPROVED** [review]

**Review by ederign — APPROVED** [review]
